test/html.test.js

```diff
@@ -150,6 +261,185 @@ describe('html', function() {
       const elem = html.toHTML(jml);
       assert.equal(elem.outerHTML, '<span data-foo="bar">hello <strong>world</strong></span>');
     });
+
+    // Also tested within the patch() suite
+    describe('element attributes', function() {
+      it('should be able to update element attributes', function() {
+        const attrs = {
+          id: 'my-el-1',
+          name: 'my-best-el',
+          className: 'foo bar baz',
+          disabled: true,
+        };
+        const jml = ['input', attrs];
+        const elem = html.toHTML(jml);
+        assert.equal(elem.outerHTML, `<input id="${attrs.id}" name="${attrs.name}" class="${attrs.className}" disabled="">`);
+      });
+
+      it('should be able to update element styles through attributes', function() {
+        // NOTE: JsonML style attribute must provide the literal inline CSS string value.
+        // An object of DOM style properties (e.g. marginTop) will not be applied as expected.
+        const attrs = {
+          style: 'color: blue; margin-top: 25px;',
+        };
+        const jml = ['input', attrs];
+        const elem = html.toHTML(jml);
+        assert.equal(elem.outerHTML, `<input style="${attrs.style}">`);
+      });
+
+      it('should be able to update set event handlers through attributes', function(done) {
+        // NOTE: event names must be lowercase. camelCase will currently fail
+        // click to complete the test otherwise the test will fail due to timeout
+        const elem = html.toHTML(['input', { onclick: () => done() }]);
+        elem.dispatchEvent(new document.defaultView.Event('click'));
+      });
+
+      it('should set null attribute values to empty strings', function() {
+        const jml = ['input', {
+          id: null,
+          class: null,
+        }];
+        const elem = html.toHTML(jml);
+        assert.equal(elem.outerHTML, '<input id="" class="">');
+      });
+
+      it('should omit set null "value" attributes', function() {
+        const jml = ['input', {
+          value: null,
+        }];
+        const elem = html.toHTML(jml);
+        assert.equal(elem.outerHTML, '<input>');
+      });
+
+      it('should set boolean attributes', function() {
+        const jml = ['input', {
+          disabled: true,
+          hidden: false,
+        }];
+        const elem = html.toHTML(jml);
+        assert.equal(elem.disabled, true);
+        assert.equal(elem.hidden, false);
+        assert.equal(elem.outerHTML, '<input disabled="">');
+      });
+
+      it('should coerce date objects to strings setting attribute values', function() {
+        const someDate = new Date('2015-06-15');
+        const jml = ['div', {
+          'data-date': someDate,
+        }];
+        const elem = html.toHTML(jml);
+        assert.equal(elem.outerHTML, `<div data-date="${someDate.toString()}"></div>`);
```

Comment:
This is helpful to demonstrate what happens. If used for the value of `<time datetime={...}` you'll find that the format is not as desired and it will be better to provide a string.